src/libImaging/TiffDecode.c

```diff
@@ -235,36 +234,97 @@ int ImagingLibTiffDecode(Imaging im, ImagingCodecState state, UINT8* buffer, int
 		}
 	}
 
-	size = TIFFScanlineSize(tiff);
-	TRACE(("ScanlineSize: %d \n", size));
-	if (size > state->bytes) {
-		TRACE(("Error, scanline size > buffer size\n"));
-		state->errcode = IMAGING_CODEC_BROKEN;
-		TIFFClose(tiff);
-		return -1;
-	}
+    TIFFSetField(tiff, TIFFTAG_JPEGCOLORMODE, JPEGCOLORMODE_RGB);
 
-	// Have to do this row by row and shove stuff into the buffer that way,
-	// with shuffle.  (or, just alloc a buffer myself, then figure out how to get it
-	// back in. Can't use read encoded stripe.
+	if (TIFFIsTiled(tiff)) {
+	    uint32 x, y, tile_y;
```

Comment:
Historical most C files have tabs and spaces mix. Please, use only four spaces as indentation in a new code (from line 239: `if (TIFFIsTiled(tiff))` to line 294: `} else {` here).

Reply:
If you want, you can create another PR with tabs to spaces conversion in `TiffDecode.c` or in the whole codebase.